WorkloadManagementSystem/PilotAgent/PilotLogger/PilotLogger.py

```diff
@@ -0,0 +1,231 @@
+""" Pilot logger module for the remote loggin system
+    Only functions send and connect are depended on the client library used.
+    The current implementation uses stomp.
+"""
+
+__RCSID__ = "$Id$"
+
+import stomp
+import sys
+import Queue
+#Since PilotLogger can be used without DIRAC relaying only on PilotLoggerTools, we
+#check two possiblities:
+try:
+  from DIRAC.WorkloadManagementSystem.PilotAgent.PilotLogger.PilotLoggerTools import generateDict, encodeMessage
+  from DIRAC.WorkloadManagementSystem.PilotAgent.PilotLogger.PilotLoggerTools import generateTimeStamp
+  from DIRAC.WorkloadManagementSystem.PilotAgent.PilotLogger.PilotLoggerTools import isMessageFormatCorrect
+  from DIRAC.WorkloadManagementSystem.PilotAgent.PilotLogger.PilotLoggerTools import readPilotLoggerConfigFile
+except ImportError:
+  from PilotLoggerTools import generateDict, encodeMessage
+  from PilotLoggerTools import generateTimeStamp
+  from PilotLoggerTools import isMessageFormatCorrect
+  from PilotLoggerTools import readPilotLoggerConfigFile
+
+def connect(host_and_port, ssl_cfg):
+  """ Connects to RabbitMQ and returns connection
+      handler or None in case of connection down.
+      Stomp-depended function.
+  """
+  try:
+    connection = stomp.Connection(host_and_ports=host_and_port, use_ssl = True)
+    connection.set_ssl(for_hosts=host_and_port,
+                       key_file = ssl_cfg['key_file'],
+                       cert_file = ssl_cfg['cert_file'],
+                       ca_certs = ssl_cfg['ca_certs'])
+    connection.start()
+    connection.connect()
+    return connection
+  except stomp.exception.ConnectFailedException:
+    print 'Connection error'
+    return None
+  except IOError:
+    print 'Could not find files with ssl certificates'
+    return None
+
+def send(msg ,destination, connect_handler):
+  """Sends a message and prints info on the screen.
+     Stomp-depended function.
+  """
+  if not connect_handler:
+    return False
+  connect_handler.send(destination=destination,
+                       body=msg)
+  print " [x] Sent %r " % (  msg )
+  return True
+
+def disconnect(connect_handler):
+  """Disconnects.
+     Stomp-depended function.
+  """
+  connect_handler.disconnect()
+
+
+def getPilotUUIDFromFile( filename = 'PilotAgentUUID' ):
+  """ Retrives Pilot UUID from the file of given name.
+  Returns:
+    str: empty string in case of errors.
+  """
+
+  try:
+    with open ( filename, 'r' ) as myFile:
+      uniqueId = myFile.read()
+    return uniqueId
+  except IOError:
+    print 'Could not open the file!!!'
+    return ""
+
+def eraseFileContent( filename ):
+  """ Erases the content of a given file.
+  """
+
+  with open(filename, 'r+') as myFile:
```

Comment:
eventhough r+ works, isn't it better to write 'w+', just because truncating a file with 'r' in the open call is weird ? :D